nixos/modules/services/security/paretosecurity.nix

```diff
@@ -38,9 +57,64 @@ in
     # if one is installed.
     # The `paretosecurity-user` timer service that is configured lower has
     # the same need.
-    systemd.services.paretosecurity.serviceConfig.Environment = [
-      "PATH=${config.system.path}/bin:${config.system.path}/sbin"
-    ];
+    systemd.services = {
+      paretosecurity.serviceConfig.Environment = [
+        "PATH=${config.system.path}/bin:${config.system.path}/sbin"
+      ];
+    }
+    // (
+
+      # Each user can set their inviteID, which creates a systemd service
+      # that runs `paretosecurity link ...` to link their device to Pareto Cloud.
+      lib.mapAttrs' (
+        username: userConfig:
+        lib.nameValuePair "paretosecurity-link-${username}" {
+          description = "Link Pareto Desktop to Pareto Cloud for user ${username}";
+          after = [ "network-online.target" ];
+          wants = [ "network-online.target" ];
+
+          serviceConfig = {
+            Type = "oneshot";
+            RemainAfterExit = true;
+            User = username;
+            StateDirectory = "paretosecurity/${username}";
+
+            ExecStart = pkgs.writeShellScript "paretosecurity-link-${username}" ''
+              set -euo pipefail
+
+              INVITE_ID="${userConfig.inviteId}"
```

Comment:
Could it be that the `inviteId` is something that one wouldn't want to be exposed in the nix expression or the nix store? If not, feel free to ignore this comment. However, if the `inviteId` is a relatively sensitive piece of information, then could you add the ability to read the `inviteId` from another source, like a file at a user-provided location, or maybe by invoking a script that fetches the `inviteId`, or something like that?

Reply:
It's not terribly sensitive, no.